src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -82,6 +82,73 @@ GTCodeGen::IntervalDefinitions::IntervalDefinitions(const Stencil& stencil)
                               Interval::makeCodeGenName(stencilFun->getInterval()));
 }
 
+/// @brief The StencilFunctionReader class parses a stencil function that is used as a boundary
+/// condition into it's stringstream. In order to use stencil_functions as boundary conditions, we
+/// need them to be memnbers of the stencil-wrapper class. The goal is to template the function s.t
+/// every field is a template argument.
+class StencilFunctionReader : public ASTStencilDesc {
```

Comment:
Given the doc, a name like `StencilFunctionAsBCGenerator` would sound more intittuive to me. This is specific visitor for BC, so I think would be good to have a name that reflects it generates BC code